Source/Core/AudioCommon/AudioCommon.cpp

```diff
@@ -53,44 +48,50 @@ static std::unique_ptr<SoundStream> CreateSoundStreamForBackend(std::string_view
 void InitSoundStream()
 {
   std::string backend = Config::Get(Config::MAIN_AUDIO_BACKEND);
-  g_sound_stream = CreateSoundStreamForBackend(backend);
+  std::unique_ptr<SoundStream> sound_stream = CreateSoundStreamForBackend(backend);
 
-  if (!g_sound_stream)
+  if (!sound_stream)
   {
     WARN_LOG_FMT(AUDIO, "Unknown backend {}, using {} instead.", backend, GetDefaultSoundBackend());
     backend = GetDefaultSoundBackend();
-    g_sound_stream = CreateSoundStreamForBackend(GetDefaultSoundBackend());
+    sound_stream = CreateSoundStreamForBackend(backend);
   }
 
-  if (!g_sound_stream || !g_sound_stream->Init())
+  if (!sound_stream || !sound_stream->Init())
   {
     WARN_LOG_FMT(AUDIO, "Could not initialize backend {}, using {} instead.", backend,
                  BACKEND_NULLSOUND);
-    g_sound_stream = std::make_unique<NullSound>();
-    g_sound_stream->Init();
+    sound_stream = std::make_unique<NullSound>();
+    sound_stream->Init();
   }
+
+  Core::System::GetInstance().SetSoundStream(std::move(sound_stream));
```

Comment:
Is there a reason you went with this over say changing this  function to `CreateSoundStream()` that returns the sound stream and that is done in `System`?

Not sure if one is really better than the other, just curious if you thought about that or had a preference.  I guess the above means more refactoring and maybe we need to split up the initialization rather than having it all in one spot.